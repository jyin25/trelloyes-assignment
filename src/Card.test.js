import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import STORE from './store';
import Card from './Card';

describe('Card component', () => {

  it ('render without issue', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Card />, div);
    ReactDOM.unmountComponentAtNode(div);
  });


  it ('render without issue', () => {
    const div = document.createElement('div');
    const allCards = {
      'a': { id: 'a', title: 'First card', content: 'lorem ipsum' }
    }
    ReactDOM.render(<Card title={allCards.a.title} content={allCards.a.content}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it ('render with data', () => {
    const cardIds = ['b', 'c', 'd', 'f', 'h', 'i', 'k'];

    const card = cardIds.map(x => STORE.allCards[x]);
    const tree = renderer
      .create(card.map(x => <Card title={x.title} content={x.content}/> ))
      .toJSON();
      expect(tree).toMatchSnapshot();
  });
})