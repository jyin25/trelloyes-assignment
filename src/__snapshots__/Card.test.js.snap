// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component render with data 1`] = `
Array [
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Second card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>,
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Third card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>,
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Fourth card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>,
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Sixth card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>,
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Eighth card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>,
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Ninth card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>,
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Eleventh card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>,
]
`;
